### ISSUE TITLE
Right click does not works with USB mouse on Android

### ISSUE THREAD

**Issue by kulturguda** [vcmi/vcmi#1588]
I installed vcmi on my Android and connected an usb mouse. In the app the right click does nothing, I cannot get infos from objects. Is there any solution to fix it?

**Comment by nullkiller** [comment]
Older version of 1.1 or 1.0 may handle mouse better. At least Bluetooth mouse. Newer one lost this feature due to SDL upgrade. Need to investigate it once again.

**Comment by kulturguda** [comment]
works with 1.1.0

**Comment by kambala-decapitator** [comment]
let's keep it open until it's actually fixed

**Comment by nullkiller** [comment]
> let's keep it open until it's actually fixed

Yes for sure. Last time I used Accross software to emulate BT mouse for testing. MAybe will try it once again somewhere in release 1.2. This functionality was in old java motion listeners.

**Comment by Laserlicht** [comment]
@IvanSavenko tested with latest nightly. With bluetooth mouse. Works fine. :)

**Comment by IvanSavenko** [comment]
Cool.
Kind of curios whether bluetooth keyboard will also work, e.g. for hotkeys.

**Comment by kulturguda** [comment]
I tested too, does not work :(

**Comment by IvanSavenko** [comment]
@kulturguda Mouse? Bluetooth one? What device / Android version do you have?
Reopening then I guess.

**Comment by kulturguda** [comment]
> then

USB wired mouse on Android 13 (OxygenOS 13.1) with OnePlus 9 Pro. This mouse works perfectly on the system and other apps (right click too).

**Comment by kulturguda** [comment]
> @kulturguda Mouse? Bluetooth one? What device / Android version do you have? Reopening then I guess.

USB wired mouse on Android 13 (OxygenOS 13.1) with OnePlus 9 Pro. This mouse works perfectly on the system and other apps (right click too).

**Comment by Laserlicht** [comment]
Okay. That's sad. Looks specific problem of device/OS version. :(

Tested again with simple USB Mouse. Seems to work everything (also right click). Like with bluetooth.
Tested on:
- Samsung Galaxy Tab S6 Lite (2020) - Android 13
- Huawai P30 - Android 10

Only drawbacks:
- If you want to scroll on borders the notification bar of android shows up (can be a little bit annoying)
- If you start using a bluetooth mouse while running it crashes (not the case if you start already with the mouse)
- you can not enable scrolling with left mouse in the GUI

**Comment by Laserlicht** [comment]
See demo here below.

The mouse and keyboard (and also the light) are connected over a (passive) hub with the tablet.

The cool thing on android is that much usb devices that are supported by linux works. I don't have a bluetooth keyboard. But this should also work.

In the past I also tested usb network adapters, cd drives, even (just for fun) an old floppy drive.

https://github.com/vcmi/vcmi/assets/13953785/1cf460f6-b9db-4217-a512-0553170c531f

**Comment by nullkiller** [comment]
you can not enable scrolling with left mouse in the GUI - likely you can change unsupported settings manually in settings.json or copy file from desktop installation. The engine is mostly the same. Only launcher differs

**Comment by GeorgeK1ng** [comment]
Right click with USB Bluetooth mouse works on Google TV with Android 11 in VCMI 1.7
